### PR TITLE
fix(ui): show conversion % and yield % in the outcome preview (#598)

### LINKED ISSUE #598: Preview of outcomes does not display yield and conversion values
It's from AC#4 https://github.com/open-reaction-database/ord-app/issues/53:
The system shows all outcomes with their respective molecular structure drawings including top label (reaction time and the **conversion %** if available) and bottom label (**yield %** and reaction role if available under each chemical component).

## Steps to Reproduce 
1.  Add a reaction with reaction time, conversion, yield and reaction role:
Image
Image
Image
2. Observe the preview
 
## Actual Result 
-  There is no conversion, yield on preview
Image

## Expected Result 
-  All mentioned paramatres should be displayed on preview

## Environment 
- Reproduced in the local environment with main branch, commit   e6b3d7adeb58ebc2dcd481cc2883d7c636af07ba;
- dev-z4acb31kcl4prqtw.us.auth0.com auth tenant;
- Desktop, Windows 11 OS, Chrome Browser.

### PR DESCRIPTION
**Pull request by skearnes**
## Summary

Per **AC#4 of #53** (#598): the outcome preview should show the limiting-reactant **conversion %** (top label) and each product's **yield %** (bottom label). Currently only reaction time and reaction role are shown.

### Changes
- **`ReactionOutcomePreview`** — the "Outcome" badge now appends the conversion next to the reaction time, e.g. **`Outcome (2 h, 95% conversion)`**, from `outcome.conversion`.
- **`ComponentMetadata`** (the product bottom label) — products now show **`85% yield`**, from the product's `YIELD` measurement.
- **`getProductYieldPercent`** helper extracts the percent value of a product's `YIELD` measurement (`ProductMeasurementType.YIELD`, Percent value).

### 🎨 Format to confirm (review)
The AC specifies the *content and position* (conversion % top, yield % bottom); I used `"<n>% conversion"` and `"<n>% yield"` as the wording. Easy to adjust to whatever matches the design mockups — flag it if you'd prefer different phrasing/precision/placement.

## Test plan
- [x] `npx tsc -b` clean
- [x] `npx vitest run` — new `getProductYieldPercent` tests (returns the YIELD percent; undefined for no-YIELD / non-percent / null / no-measurements) and a `ComponentMetadata` test asserting `85% yield` renders. Full suite green.
- [x] `prettier` / `eslint` clean
- [ ] Held for review — behavioral preview change with a format choice; runtime setup (a reaction with conversion + a YIELD measurement) is awkward to script, and the data extraction is unit-covered.

Closes #598.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR surfaces conversion % and yield % in the reaction outcome preview, addressing AC#4 of #598. Data extraction is cleanly isolated in a new `getProductYieldPercent` helper, and the UI changes are minimal and well-guarded.

- **`getProductYieldPercent`** walks the product's `measurements` array for a `YIELD` entry, type-narrows to `ReactionMeasurementValueType.Percent`, and guards the `OrdOptional<number>` value with `!= null` before returning it — all edge cases (no measurements, non-percent type, null value) are covered by dedicated unit tests.
- **`ReactionOutcomePreview`** builds the outcome badge label by composing `outcomeTime` and `conversion` with `filter(Boolean)`, so the parenthetical is omitted entirely when both are absent.
- **`ComponentMetadata`** renders `{productYield}% yield` only for components that carry a `measurements` field (`ReactionProduct`), preserving the existing display for `ReactionInputComponent`.

<h3>Confidence Score: 5/5</h3>

Safe to merge — changes are additive display-only additions with no mutations to shared state or data-fetching paths.

All data access is guarded against null/undefined at every layer, the new helper is fully unit-tested, and the UI refactor in ComponentMetadata is a direct logical equivalent of the original inline expressions.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| ui/src/common/components/ReactionPreview/reactionPreview.utils.ts | Adds getProductYieldPercent helper; correctly uses optional chaining and != null guard against OrdOptional<number> |
| ui/src/common/components/ReactionPreview/ReactionOutcomePreview.tsx | outcomeLabel memo correctly composes time + conversion parts with filter(Boolean), dependency array is complete |
| ui/src/common/components/ReactionPreview/ComponentMetadata.tsx | productYield != null guard is correct for number | undefined; amount/isLimiting refactor is logically equivalent to original |
| ui/src/common/components/ReactionPreview/reactionPreview.utils.test.ts | Five cases cover the happy path, no-YIELD, non-percent, null value, and missing measurements; comprehensive |
| ui/src/common/components/ReactionPreview/ComponentMetadata.test.tsx | New render test correctly asserts '85% yield' text is present for a product with a YIELD measurement |

</details>

</details>

<sub>Reviews (2): Last reviewed commit: ["style(ui): match file-local import conve..."](https://github.com/open-reaction-database/ord-app/commit/b19f67ba327efdd31134175eaf7bc606e6807061) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=37455554)</sub>

### CODE REVIEW

File: ui/src/common/components/ReactionPreview/ComponentMetadata.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ import type {
 } from 'store/entities/reactions/reactionComponent/reactionComponent.types';
 import { ReactionBoolean } from 'store/entities/reactions/reactionEntity/reactionEntity.types';
 import { renderValuePrecisionUnit } from 'features/reactions/ReactionView/renderValuePrecisionUnit';
+import { getProductYieldPercent } from 'common/components/ReactionPreview/reactionPreview.utils.ts';
```
[the remark this answers is not in the document]
Comment (skearnes) [reply ui/src/common/components/ReactionPreview/ComponentMetadata.tsx]: Fixed — dropped the `.ts` extension to match the other imports in this file.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
